### PR TITLE
Instructor: courses page: show creation time in active/archived courses #1297

### LINKED ISSUE #1297: Instructor: courses page: show creation time in active/archived courses
We can show the creation date as a `Created on` column and make it sortable.
Can be done for both 'active' and 'archived' courses.

### PR DESCRIPTION
**Pull request by HirdayGupta**
Fixes #1297 

**Outline of solution:**
- Added a method in CourseAttributes to get createdAt date field as a formatted String.
- Added this string as data to active and archived row classes
- Added this new data to the HTML tags.
- Used the `sort-comparator` from `toggle-sort` to sort by date.

Note: I have formatted the date using the `TimeHelper.formatDateTimeForInstructorHomePage` method. I had originally simply used the `formatDate` method but that was neither as user-readable and universally understood as this nor was it being parsed by JS as a date object (for `sortDate` comparator to work). Let me know if I should change the method name to `formatDateTimeForInstructorHomePageAndCoursesPage`.

**TODO:**
- ~~Run god mode.~~

### CODE REVIEW

**Comment by HirdayGupta** [comment]
Note: Changed the date format again in 3f9bd1b. This is because I realised that the previously used format was not displaying year information, and unlike sessions in the InstructorHomePage, instructors might want to see year information at a glance in this case. The new format also removes the `hh:mm` information from the time displayed because, in my opinion, the exact creation time of the course doesn't matter too much. Let me know if you disagree. I could use `formatTime12H` to include time information as well.

The new date format is still JS parsable, and hence `sortDate` still works as expected.

Also, realised that some tests were failing because the creation time was the same as the current time (hence expected HTML was unreliable). So I updated HtmlHelper to replace the current time in the new format with a placeholder and also updated the related test HTML files so that HtmlHelperTest and GodmodeTest will check this new replacement.

**Comment by HirdayGupta** [comment]
Ready for review.

**Comment by HirdayGupta** [comment]
@damithc Can label `s.ToReview`?

**Comment by HirdayGupta** [comment]
@whipermr5 I assume you mean adding in the date as a data attribute in ISO 8601 format. There's just one problem for this: the timezone info in the CourseAttributes object contains the timezone information as a String and not a double. As a result, I am unable to use the `convertLocalDateToUtc` method we had created in TimeHelper a while back (in #7740). ANy suggestions for this? Maybe I can make use of the `TIME_ZONE_CITIES_MAP` in TimeHelper?

**Comment by whipermr5** [comment]
I believe the time data in `CourseAttributes` is already in UTC.

**Comment by HirdayGupta** [comment]
@whipermr5 Ready for review :)

**Comment by HirdayGupta** [comment]
@whipermr5 Please review.

**Comment by HirdayGupta** [comment]
@whipermr5 Review reminder please.

**Comment by HirdayGupta** [comment]
@whipermr5 Please review soon. :)

**Comment by whipermr5** [comment]
@HirdayGupta So sorry! Will review by tonight 😅

File: src/main/java/teammates/ui/pagedata/InstructorCoursesPageData.java [hunk 1]
```diff
@@ -100,6 +100,12 @@ private ArchivedCoursesTable convertToArchivedCoursesTable(List<CourseAttributes
 
             ArchivedCoursesTableRow row = new ArchivedCoursesTableRow(SanitizationHelper.sanitizeForHtml(course.getId()),
                                                                       SanitizationHelper.sanitizeForHtml(course.getName()),
+                                                                      SanitizationHelper.sanitizeForHtml(
+                                                                              course.getCreatedAtDateString()),
+                                                                      SanitizationHelper.sanitizeForHtml(
+                                                                              course.getCreatedAtDateStamp()),
+                                                                      SanitizationHelper.sanitizeForHtml(
+                                                                              course.getCreatedAtFullDateTimeString()),
```
Comment (whipermr5) [inline src/main/java/teammates/ui/pagedata/InstructorCoursesPageData.java]: I believe these 3 fields don't need sanitisation as they are safe strings not input by the user but generated by the system from a date object.

File: src/main/java/teammates/ui/pagedata/InstructorCoursesPageData.java [hunk 2]
```diff
@@ -152,6 +158,12 @@ private ActiveCoursesTable convertToActiveCoursesTable(List<CourseAttributes> co
 
             ActiveCoursesTableRow row = new ActiveCoursesTableRow(SanitizationHelper.sanitizeForHtml(course.getId()),
                                                                   SanitizationHelper.sanitizeForHtml(course.getName()),
+                                                                  SanitizationHelper.sanitizeForHtml(
+                                                                          course.getCreatedAtDateString()),
+                                                                  SanitizationHelper.sanitizeForHtml(
+                                                                          course.getCreatedAtDateStamp()),
+                                                                  SanitizationHelper.sanitizeForHtml(
+                                                                          course.getCreatedAtFullDateTimeString()),
```
Comment (whipermr5) [inline src/main/java/teammates/ui/pagedata/InstructorCoursesPageData.java]: Same here.

File: src/main/java/teammates/ui/pagedata/InstructorCoursesPageData.java (hunk 1 above)
Comment (HirdayGupta) [inline src/main/java/teammates/ui/pagedata/InstructorCoursesPageData.java]: Great catch as always. Committed the updates. :)

**Comment by HirdayGupta** [comment]
@whipermr5 Changes made. Ready for review!

**Comment by HirdayGupta** [comment]
@whipermr5 Shouldn't take too much of your time now that initial review is over. Please look at this and see if it is merge worthy :)

File: src/main/webapp/WEB-INF/tags/instructor/course/activeCoursesTable.tag [hunk 3]
```diff
@@ -12,6 +12,9 @@
       <th id="button_sortcoursename" class="button-sort-none toggle-sort">
         Course Name<span class="icon-sort unsorted"></span>
       </th>
+      <th id="button_sortcoursecreateddate" data-toggle-sort-comparator="sortDate" data-toggle-sort-extractor="dateStampExtractor" class="button-sort-none toggle-sort">
+        Course Created At<span class="icon-sort unsorted"></span>
```
Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/instructor/course/activeCoursesTable.tag]: Use `Creation Date` for consistency with the sessions table. The word `Course` can be dropped for brevity.

File: src/main/webapp/WEB-INF/tags/instructor/course/archivedCoursesTable.tag [hunk 4]
```diff
@@ -14,13 +14,23 @@
       <th id="button_sortid" class="button-sort-none toggle-sort">
         Course Name<span class="icon-sort unsorted"></span>
       </th>
+      <th id="button_sortcoursecreateddate" data-toggle-sort-comparator="sortDate" data-toggle-sort-extractor="dateStampExtractor" class="button-sort-none toggle-sort">
+        Course Created At<span class="icon-sort unsorted"></span>
```
Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/instructor/course/archivedCoursesTable.tag]: Same here.

**Comment by HirdayGupta** [comment]
@whipermr5 changed the table heading to `Creation Date`. Apologies for the long list of commits, I am away from my workspace and have made these changes on mobile as I did not wish to delay this further. Ready for review! (Hopefully Travis agrees with me!) 😅 

Apologies once again for the extraordinary number of commits for this change and for the departure from standard behaviour of using god mode to update test files (I felt the change was trivial enough). Was using Github's webapp for this and I don't think I can add a single commit with changes on multiple files in the web interface. Sincere apologies.

**Comment by HirdayGupta** [comment]
@whipermr5 travis build has passed! Please review :)

**Pull request merged**